src/test/java/seedu/fast/model/tag/TagTest.java

```diff
@@ -17,4 +25,62 @@ public void constructor_invalidTagName_throwsIllegalArgumentException() {
         assertThrows(IllegalArgumentException.class, () -> new Tag(invalidTagName));
     }
 
+    @Test
+    public void isValidTagName() {
+        // null tag name
+        assertThrows(NullPointerException.class, () -> Tag.isValidTagName(null));
+    }
+
+    @Test
+    public void isValidTagTerm() {
+        //null tag term
+        assertThrows(NullPointerException.class, () -> Tag.isValidTagTerm(null));
+
+        //valid Tag term
+        assertTrue(Tag.isValidTagTerm(VALID_TAG_TERM));
+
```

Comment:
I think that testing for valid tag terms could be more comprehensive, since currently, tag names are alphanumeric.

Reply:
Good suggestion, can be added with a quick commit.